Updated documentation

MikeTschudi opened the pull request.

previnWong commented:
I like ```build.bat``` runs everything for the end user.  But something that I found lacking in the experience was a "call to action".  Either in the document or as a person that just finished running build.bat.  

Maybe we should have a website that loads when build.bat finishes that open a page with links to all the demos so a user can test the demos right away.  I understand that there is no webserver being spun up to serve a page, but maybe load it from system path is fine?

Or have in the document a path to a global demo page that has all the demos a user could run and try.

MikeTschudi commented:
> ![image](https://private-user-images.githubusercontent.com/11183838/360695410-7ded916c-86e8-4ddb-a12e-f37f5d7462d5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzNjcyMzUsIm5iZiI6MTcyNDM2NjkzNSwicGF0aCI6Ii8xMTE4MzgzOC8zNjA2OTU0MTAtN2RlZDkxNmMtODZlOC00ZGRiLWExMmUtZjM3ZjVkNzQ2MmQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIyVDIyNDg1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU0NDEyNGFlYmM0MWY4MDFiOWViYmJiMGQ4ZDdlYWFjYmI5NzgwY2Y5YzA5YmY5Nzk1MzYzM2NjZjZmMTI2Y2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2HMROuaof7vIe4-qaQfCNWu91xVF57_8DYr7hcaOsH0)
> 
> The description of things is alittle confusing, I guess because it uses the same words to describe things. I understand an `item` becomes a `template`. Then is `Solution Item` an actual thing `item` in an AGO org?
> 
> The also, is `Item Templates` just really `template`(s) or is this something else?

It is awkward that we're using the same item and term to describe two ideas.  How about this version?

---
A `Solution ArcGIS Online Item` can take two forms:

1. A `Solution Template` resides in a host ArcGIS Online organization. It contains a list of templates of ArcGIS Online items that work together. It is used to deploy a Solution and its items into destination ArcGIS Online organizations.
2. A `Deployed Solution` resides in a destination ArcGIS Online organization. It is created when a Solution is deployed and contains a list of references to items deployed for a Solution; it serves as a table of contents for the deployment. This grouping of references also simplifies deleting the deployed Solution.

previnWong commented on README.md:
```diff
@@ -96,43 +96,41 @@ For a list of all available commands run `npm run`.
-  * postinstall runs `bootstrap`
-  * bootstrap
-  * precommit
+  * precommit lints staged files
```
pre-commit I think is how it is written on the interwebs

previnWong commented on README.md:
```diff
@@ -96,43 +96,41 @@ For a list of all available commands run `npm run`.
 * publishing doc
   * `npm run docs:build` builds the documentation ___(note that this script creates a `docs` folder, deleting any existing one)___
   * `npm run docs:deploy` pushes the documentation to the repository's gh-pages
+  * `npm run docs:srihash` generates the srihashes
```
`npm run docs:srihash` generates the srihashes.  Learn more about SRI Hash here https://www.srihash.org/
 
 I had no idea what SRI hash was, so figure a link to it would be good.

previnWong commented on README.md:
```diff
@@ -96,43 +96,41 @@ For a list of all available commands run `npm run`.
 
 * testing
   * `npm run test` lints, then runs `test:chrome` tests to confirm that the API is functioning as expected
-  * `npm run test:browsers` runs karma in the Chrome, Firefox, and Chromium Edge browsers
```
`npm run test:browsers` runs karma in the Chrome, Firefox, and Chromium Edge browsers. Learn more about Karma here  https://karma-runner.github.io/6.4/config/browsers.html

previnWong reviewed (approved):
minor stuff that came to mind as I went through it.

MikeTschudi commented:
Thank you for the comments, @previnWong

I updated README.md and Getting Started.md for
* description of a Solution item
* building to the point to running the demos
* unit testing on Edge and Safari
* pre-commit fix
* srihash link
* restored testing on Edge
* added doc about adding a new package

Other:
* Testing on Safari: I haven't succeeded in building it on MacOS; `build.sh` works on Linux systems, but not MacOS.  I added the npm command because Karma supports the browser, but the build needs to be resolved.
* Due to permissions required by a website, it's not possible for the build script to launch a welcome site. But I added how to set up the demos list site to Getting Started.
* We don't want to support a public demos page, so I just added instructions for one's own demos page
* I'm open to learning more whether we should not write out both esm & cjs the first time
* I'm open to adding documentation about hot file loading & npm run watch

previnWong reviewed (approved):
I'm good with the changes.
